### ISSUE TITLE
Intel 2022.0.0 unconvinced that string literals are `constexpr` in `perf_test`

### ISSUE THREAD

**Issue by cwpearson** [kokkos/kokkos-kernels#2056]
- [ ]I'm not sure why only the `perf_test` binaries have this problem.
- [ ] Kokkos reports the compiler version is `2022.0.0` but lmod says `2022.1.2`.

On Blake:
```
export ROOT_DIR=<put something here>

export KOKKOS_SHA=f8788ef2ae1940b627cc6ebc6abeef2c34e7e8dc # 2023 11 30
export KOKKOS_SRC="$ROOT_DIR/kokkos-${KOKKOS_SHA:0:8}"
export KOKKOS_BUILD="$ROOT_DIR/kokkos-build-${KOKKOS_SHA:0:8}"
export KOKKOS_INSTALL="$ROOT_DIR/kokkos-install-${KOKKOS_SHA:0:8}"

export KERNELS_SHA=a80eb9114ddda2d9454e4f3cc8a3dd5143ecdfc8 # 2023 11 30
export KERNELS_SRC="$ROOT_DIR/kernels-${KERNELS_SHA:0:8}"
export KERNELS_BUILD="$ROOT_DIR/kernels-build-${KERNELS_SHA:0:8}"

export MODULEPATH="$MODULEPATH:/projects/x86-64/modulefiles"
module load intel/oneAPI/hpc-toolkit/2022.1.2
module load cmake

# intel blows up SSH for some reason?
module del intel/oneAPI/hpc-toolkit/2022.1.2
git clone git@github.com:kokkos/kokkos.git "$KOKKOS_SRC" || true
(cd "$KOKKOS_SRC" && git checkout $KOKKOS_SHA) || true
git clone git@github.com:kokkos/kokkos-kernels.git "$KERNELS_SRC" || true
(cd "$KERNELS_SRC" && git checkout $KERNELS_SHA) || true

# re-set up our environment
module load intel/oneAPI/hpc-toolkit/2022.1.2
module load cmake

## Configure Kokkos
cmake -S "$KOKKOS_SRC" -B "$KOKKOS_BUILD" \
-DCMAKE_INSTALL_PREFIX="$KOKKOS_INSTALL" \
-DCMAKE_CXX_STANDARD=17 \
-DCMAKE_BUILD_TYPE=Release \
-DCMAKE_CXX_COMPILER=icpx \
-DKokkos_ENABLE_OPENMP=ON \
-DKokkos_ARCH_SPR=ON

## Build & Install Kokkos
cmake --build "$KOKKOS_BUILD" -j "$(nproc)" -t install

## Configure Kernels
cmake -S "$KERNELS_SRC" -B "$KERNELS_BUILD" \
-DKokkos_DIR="$KOKKOS_INSTALL/lib64/cmake/Kokkos" \
-DCMAKE_BUILD_TYPE=Release \
-DCMAKE_CXX_COMPILER=icpx \
-DKokkosKernels_ENABLE_TPL_MKL=ON \
-DKokkosKernels_ENABLE_TESTS=ON \
-DKokkosKernels_ENABLE_PERFTESTS=ON \
-DKokkosKernels_ENABLE_BENCHMARK=ON

## Build Kernels
VERBOSE=1 cmake --build "$KERNELS_BUILD" -j "$(nproc)"
```

Yields errors like this (only in `perf_test`)

```
In file included from .../perf_test/BenchmarkMain.cpp:21:
In file included from .../perf_test/Benchmark_Context.hpp:31:
.../KokkosKernels_Version_Info.hpp:25:28: error: constexpr variable 'GIT_BRANCH' must be initialized by a constant expression
constexpr std::string_view GIT_BRANCH       = R"branch(HEAD)branch";
```

A possible fix is to make `KokkosKernels_Version_Info.hpp` look like this:

```c++
namespace KokkosKernels {
namespace Impl {

using namespace std::string_view_literals; // add this

constexpr std::string_view GIT_BRANCH       = R"branch(HEAD)branch"sv; // add sv
...

}  // namespace Impl
}  // namespace KokkosKernels
```

**Comment by ndellingwood** [comment]
@cwpearson @eeprude I was curious and hopped on blake to check what was loaded with the intel compiler:

```
[ndellin@blake ~]$ export MODULEPATH="$MODULEPATH:/projects/x86-64/modulefiles"
[ndellin@blake ~]$ module load intel/oneAPI/hpc-toolkit/2022.1.2
[ndellin@blake ~]$ module list

Currently Loaded Modules:
  1) binutils/2.30.0   2) gcc/7.2.0   3) intel/oneAPI/base-toolkit/2022.1.2   4) intel/oneAPI/hpc-toolkit/2022.1.2
```

gcc/7.2.0 does not support c++17 , can you try swapping for a newer gcc (e.g. ` module swap gcc/7.2.0 gcc/8.3.0`) ? I'm not certain how this will interact with the toolchain, but it is worth a try. In any case, using the compiler with gcc/7.2.0 in the toolchain is not expected to work since the 4.0 release

**Comment by ndellingwood** [comment]
Also, is there any reason to use intel/2022.1.2 (which I think is part of the older module system before blake's DST) as opposed to one of the recent installs like `intel-oneapi-compilers/2023.1.0`?

**Comment by lucbv** [comment]
I would also recommend the newer intel compiler, I believe `2023.2.0` is actually the default so why not use that? It has worked well for me so far : )

**Comment by eeprude** [comment]
@lucbv A "module avail" at caraway does not show any 2023.2.0 available.

**Comment by ndellingwood** [comment]
@eeprude we were referring to blake. You should be able to check for modules with spider (e.g. `module spider intel`)

**Comment by eeprude** [comment]
Sorry... My bad! Let me try again.

**Comment by ndellingwood** [comment]
@eeprude there are a few different module systems available, if spider does not work or you don't see e.g. 2023.2.0 or 2023.1.0, you may need to run `source /projects/x86-64-icelake-rocky8/spack-config/blake-setup-user-module-env.sh`

**Comment by cwpearson** [comment]
Seems like this was an issue with accidentally using an unsuppoed gcc stdlib or something.
